src/main/java/com/ldtteam/domumornamentum/block/decorative/PillarBlock.java

```diff
@@ -0,0 +1,416 @@
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.EnumProperty;
+import net.minecraft.world.level.block.state.properties.Property;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.level.material.Material;
+import net.minecraft.world.level.storage.loot.LootContext;
+import net.minecraft.world.phys.HitResult;
+import net.minecraft.world.phys.shapes.BooleanOp;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+import net.minecraftforge.registries.IForgeRegistry;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class PillarBlock extends AbstractBlock<PillarBlock> implements IMateriallyTexturedBlock, ICachedItemGroupBlock, EntityBlock
+{
+
+    public static final List<IMateriallyTexturedBlockComponent> COMPONENTS = ImmutableList.<IMateriallyTexturedBlockComponent>builder()
+        .add(new SimpleRetexturableComponent(new ResourceLocation("block/oak_planks"), ModTags.PILLAR_MATERIALS, Blocks.OAK_PLANKS))
+        .build();
+
+    private final List<ItemStack> fillItemGroupCache = Lists.newArrayList();
+    /**
+     * the shape of the column section
+     */
+    private static final Optional<VoxelShape> pillar_capital_shape = Stream.of(
+     Shapes.box(0.4375, 0, 0.1875, 0.5625, 0.75, 0.8125),
+     Shapes.box(0.4375, 0, 0.1875, 0.5625, 0.75, 0.8125),
+     Shapes.box(0.4375, 0, 0.1875, 0.5625, 0.75, 0.8125),
+     Shapes.box(0.1875, 0, 0.4375, 0.8125, 0.75, 0.5625),
+     Shapes.box(0.1875, 0, 0.4375, 0.8125, 0.75, 0.5625),
+     Shapes.box(0.1875, 0, 0.4375, 0.8125, 0.75, 0.5625),
+     Shapes.box(0.1875, 0, 0.4375, 0.8125, 0.75, 0.5625),
+     Shapes.box(0.4375, 0, 0.1875, 0.5625, 0.75, 0.8125)
+     ).reduce((v1,v2) ->Shapes.join(v1, v2,BooleanOp.OR));
+
+    /**
+     * This block's name.
+     */
+    public static final String                      BLOCK_NAME     = "blockpillar";
+
+    /**
+     * The hardness this block has.
+     */
+    private static final float                      BLOCK_HARDNESS = 3F;
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float                      RESISTANCE     = 1F;
+    /**
+     * the types of pillar shapes that can be rendered
+     */
+    private static final EnumProperty<PillarShapeType> column = EnumProperty.create("column", PillarShapeType.class);
+
+    /**
+     * base constructor
+     */
+    public PillarBlock()
+    {
+        super(BlockBehaviour.Properties.of(Material.GLASS).strength(BLOCK_HARDNESS, RESISTANCE));
```

Comment:
`Material.GLASS` is not the right choice here.
Choose `ROCK` or `WOOD` instead.